extensions/amp-analytics/0.1/cross-domain-iframe-message-queue.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from '../../../src/log';
+import {timerFor} from '../../../src/services';
+import {
+  AMP_ANALYTICS_3P_MESSAGE_TYPE,
+} from '../../../src/3p-analytics-common';
+
+/** @private @const {string} */
+const TAG_ = 'amp-analytics.CrossDomainIframeMessageQueue';
+
+/** @private @const {number} */
+const MESSAGE_THROTTLE_TIME_ = 100;
+
+/** @private @const {number} */
+const MAX_QUEUE_SIZE_ = 100;
+
+/**
+ * @visibleForTesting
+ */
+export class CrossDomainIframeMessageQueue {
+  /**
+   * Constructor
+   * @param {!../../../3p/iframe-messaging-client.IframeMessagingClient}
+   *   iframeMessagingClient Facilitates
+   * cross-frame communication
+   * @param {!string} envelopeType The type of envelope that will be used to
+   * contain the queued messages (see 3p-analytics-common)
+   */
+  constructor(iframeMessagingClient, envelopeType) {
+    /** @type {Array<(
+    *            ../../../src/3p-analytics-common.ampAnalytics3pEvent|
+    *            ../../../src/3p-analytics-common.ampAnalytics3pNewCreative
+    *          )>} */
```

Comment:
I find this preferable to a linter exception to the 80-char line, but if you feel otherwise I am happy to change it.